src/function.py

```diff
@@ -50,32 +56,35 @@ def await_function_deployment(client: CogniteClient, external_id: str, wait_time
     while time.time() <= t0 + wait_time_sec:
         function = client.functions.retrieve(external_id=external_id)
         if function is None:  # Should not ever happen... :shrug:
-            err = f"No function with external_id={external_id} exists!"
+            err = f"No function with {external_id=} exists!"
             logger.warning(err)
             raise FunctionDeployError(err)
-        elif function.status == "Ready":
+
+        elif function.status == FunctionStatus.READY:
             logger.info(f"Function deployment successful! Deployment took {precisedelta(time.time()-t0)}")
             return function
-        elif function.status == "Failed":
-            logger.warning(f"Deployment failed after {precisedelta(time.time()-t0)}! Error: {function.error['trace']}")
-            raise FunctionDeployError(function.error["trace"])
+
+        elif function.status == FunctionStatus.FAILED:
+            err_msg = f"Error message: {function.error['message']}.\nTrace: {function.error['trace']}"
+            logger.warning(f"Deployment failed after {precisedelta(time.time()-t0)}! {err_msg}")
+            raise FunctionDeployError(err_msg)
         time.sleep(5)
 
     err = f"Function {external_id} (ID: {function.id}) did not deploy within {precisedelta(wait_time_sec)}."
     logger.error(err)
     raise FunctionDeployTimeout(err)
 
 
-def try_delete(client: CogniteClient, external_id: str, remove_schedules: bool = True):
-    try_delete_function(client, external_id)
-    try_delete_function_file(client, get_file_name(external_id))
-    # Schedules live on when functions die, so we always clean up:
+def delete_single_cognite_function(client: CogniteClient, external_id: str, remove_schedules: bool):
+    delete_function(client, external_id)
+    delete_function_file(client, get_file_name(external_id))
+    # Schedules live on when functions die, so we clean them up only if specified:
     if remove_schedules:
-        delete_all_schedules_for_ext_id(client, external_id)
+        delete_function_schedules(client, external_id)
```

Comment:
what should we delete first? schedule or function? order does make sense IMO

Reply:
So currently it is:

1. Function object [Function]
2. Function file [Files]
3. Schedules (optional)

I think I'm happy with this, but also gladly willing to change the ordering.